video/out/wayland_common.c

```diff
@@ -1297,3 +1299,18 @@ bool vo_wayland_config (struct vo *vo, uint32_t flags)
 
     return true;
 }
+
+void vo_wayland_request_frame(struct vo *vo, void *data, vo_wayland_frame_cb cb)
+{
+    struct vo_wayland_state *wl = vo->wayland;
+    wl->frame.data = data;
+    wl->frame.function = cb;
+    MP_DBG(wl, "restart frame callback\n");
+    frame_callback(wl, NULL, 0);
+}
+
+void vo_wayland_wait_frame(struct vo *vo)
+{
+    struct vo_wayland_state *wl = vo->wayland;
+    wl_display_roundtrip(wl->display.display);
```

Comment:
>Would using eventfd be reasonable?

AFAIK not BSD compatible. (I don't know where Wayland is on BSD, but we made an effort to reduce Linux specifics earlier.)

But I think you can run the wayland event loop just fine. Since the vo_opengl code always calls the wayland_common/gl_wayland code, and never the other way around, there should be no problem.